Marlin/src/Marlin.cpp

```diff
@@ -719,6 +719,10 @@ void setup() {
     tmc2208_serial_begin();
   #endif
 
+  #if ENABLED(HAS_BOARD_INIT)
```

Comment:
I think `#ifdef BOARD_INIT` is probably fine. The `HAS_STUFF` macros are usually there for things that combine more than one test.